app/src/main/java/com/monkeyteam/chimpagne/ui/event/details/supplies/SuppliesScreen.kt

```diff
@@ -0,0 +1,173 @@
+
+  var displayAddPopup by remember { mutableStateOf(false) }
+  if (displayAddPopup) {
+    EditSupplyDialog(
+        supply = ChimpagneSupply(),
+        onDismissRequest = { displayAddPopup = false },
+        onSave = { eventViewModel.updateSupplyAtomically(it) })
+  }
+
+  var displayedSupply by remember { mutableStateOf(ChimpagneSupply()) }
+  var displayAssignPopup by remember { mutableStateOf(false) }
+  if (displayAssignPopup) {
+    if (eventUiState.currentUserRole == ChimpagneRole.GUEST) {
+      GuestSupplyDialog(
+          supply = displayedSupply,
+          assignMyself = {
+            if (it) {
+              eventViewModel.assignSupplyAtomically(
+                  displayedSupply.id, accountsUiState.currentUserUID!!)
+            } else {
+              eventViewModel.unassignSupplyAtomically(
+                  displayedSupply.id, accountsUiState.currentUserUID!!)
+            }
+            displayAssignPopup = false
+            displayedSupply = ChimpagneSupply()
+          },
+          loggedUserUID = accountsUiState.currentUserUID!!,
+          accounts = accountsUiState.fetchedAccounts,
+          onDismissRequest = {
+            displayAssignPopup = false
+            displayedSupply = ChimpagneSupply()
+          })
+    } else {
+      StaffSupplyDialog(
+          supply = displayedSupply,
+          updateSupply = { eventViewModel.updateSupplyAtomically(it) },
+          deleteSupply = { eventViewModel.removeSupplyAtomically(displayedSupply.id) },
+          loggedUserUID = accountsUiState.currentUserUID!!,
+          accounts = accountsUiState.fetchedAccounts,
+          onDismissRequest = {
+            displayAssignPopup = false
+            displayedSupply = ChimpagneSupply()
+          })
+    }
+  }
+
+  @Composable
+  fun DisplaySupplyListIfNotEmpty(
+      listTitle: String,
+      supplyList: List<ChimpagneSupply>,
+      testTag: String
+  ) {
+    if (supplyList.isNotEmpty()) {
+      Text(text = listTitle, modifier = Modifier.padding(12.dp, 8.dp))
+      LazyColumn(modifier = Modifier.testTag(testTag)) {
+        supplyList.forEach { supply ->
+          item {
+            SupplyCard(supply = supply) {
+              displayedSupply = supply
+              displayAssignPopup = true
+            }
+          }
+        }
+      }
+    }
+  }
+
+  Scaffold(
+      floatingActionButton = {
+        if (listOf(ChimpagneRole.OWNER, ChimpagneRole.STAFF)
+            .contains(eventUiState.currentUserRole)) {
+          FloatingActionButton(
+              onClick = { displayAddPopup = true }, modifier = Modifier.testTag("supply_add")) {
+                Icon(Icons.Default.Add, contentDescription = "Add")
+              }
+        }
+      },
+      topBar = {
+        TopAppBar(
+            title = { Text(stringResource(id = R.string.supplies_screen_title)) },
+            modifier = Modifier.shadow(4.dp),
+            navigationIcon = {
+              IconButton(onClick = { navObject.goBack() }) {
```

Comment:
The reusable component GoBackButton should be used here:

https://github.com/Chimpagne/ChimpagneApp/blob/main/app/src/main/java/com/monkeyteam/chimpagne/ui/components/Button.kt

Reply:
amazing component, thx